shop/src/main/java/ro/msg/learning/shop/domain/Location.java

```diff
@@ -0,0 +1,45 @@
+package ro.msg.learning.shop.domain;
+
+import jakarta.persistence.*;
+import lombok.*;
+import org.antlr.v4.runtime.misc.NotNull;
+import org.hibernate.annotations.UuidGenerator;
+
+import java.util.Set;
+import java.util.UUID;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "location")
+@Builder
+public class Location {
+    @OneToMany(mappedBy = "location")
+    Set<Stock> stock;
+    @Id
+    @Column(name = "id")
+    @UuidGenerator
+    private UUID locationId;
+
+    @NotNull
```

Comment:
We don't need `@NotNull` in entities in this way. This is mostly used when we generate the database from the models using hibernate. In our case we don't do that so if you want not null constraints you should write them into the migration files.